core/src/main/java/com/datastrato/gravitino/connector/authorization/AuthorizationHook.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.datastrato.gravitino.connector.authorization;
+
+import java.io.Closeable;
+
+/**
+ * authorization operations hooks interfaces. Note: Because each interface function needs to perform
+ * multiple steps in the underlying permission system, the implementation method of these function
+ * interface must be idempotent.
+ */
+public interface AuthorizationHook
```

Comment:
Should we only implement this `AuthorizationHook`, or can developers directly implement `AuthorizationUserHook`?